gcloud-java-dns/src/test/java/com/google/gcloud/dns/ITDnsTest.java

```diff
@@ -0,0 +1,1268 @@
+import java.util.LinkedList;
+import java.util.List;
+import java.util.UUID;
+import java.util.concurrent.TimeUnit;
+
+public class ITDnsTest {
+
+  // todo(mderka) Implement test for creating invalid change when DnsException is finished. #673
+
+  public static final int TIME_LIMIT = 10;
+  public static final String PREFIX = "gcldjvit-";
+  public static final Dns DNS = DnsOptions.builder().build().service();
+  public static final String PROJECT_ID = DNS.options().projectId();
+  public static final String ZONE_NAME1 = (PREFIX + UUID.randomUUID()).substring(0, 32);
+  public static final String ZONE_NAME_EMPTY_DESCRIPTION =
+      ("gcldjvit-" + UUID.randomUUID()).substring(0, 32);
+  public static final String ZONE_NAME_TOO_LONG = (PREFIX + UUID.randomUUID());
+  public static final String ZONE_DESCRIPTION1 = "first zone";
+  public static final String ZONE_DNS_NAME1 = ZONE_NAME1 + ".com.";
+  public static final String ZONE_DNS_EMPTY_DESCRIPTION = ZONE_NAME_EMPTY_DESCRIPTION + ".com.";
+  public static final String ZONE_DNS_NAME_NO_PERIOD = ZONE_NAME1 + ".com";
+  public static final ZoneInfo ZONE1 = ZoneInfo.builder(ZONE_NAME1)
+      .description(ZONE_DESCRIPTION1)
+      .dnsName(ZONE_DNS_EMPTY_DESCRIPTION)
+      .build();
+  public static final ZoneInfo ZONE_EMPTY_DESCRIPTION =
+      ZoneInfo.builder(ZONE_NAME_EMPTY_DESCRIPTION)
+          .description(ZONE_DESCRIPTION1)
+          .dnsName(ZONE_DNS_NAME1)
+          .build();
+  public static final ZoneInfo ZONE_NAME_ERROR =
+      ZoneInfo.builder(ZONE_NAME_TOO_LONG)
+          .description(ZONE_DESCRIPTION1)
+          .dnsName(ZONE_DNS_NAME1)
+          .build();
+  public static final ZoneInfo ZONE_MISSING_DESCRIPTION =
+      ZoneInfo.builder(ZONE_NAME1)
+          .dnsName(ZONE_DNS_NAME1)
+          .build();
+  public static final ZoneInfo ZONE_MISSING_DNS_NAME =
+      ZoneInfo.builder(ZONE_NAME1)
+          .description(ZONE_DESCRIPTION1)
+          .build();
+  public static final ZoneInfo ZONE_DNS_NO_PERIOD =
+      ZoneInfo.builder(ZONE_NAME1)
+          .description(ZONE_DESCRIPTION1)
+          .dnsName(ZONE_DNS_NAME_NO_PERIOD)
+          .build();
+  public static final DnsRecord A_RECORD_ZONE1 = DnsRecord
+      .builder("www." + ZONE1.dnsName(), DnsRecord.Type.A)
+      .records(ImmutableList.of("123.123.55.1"))
+      .ttl(25, TimeUnit.SECONDS)
+      .build();
+  public static final DnsRecord AAAA_RECORD_ZONE1 = DnsRecord
+      .builder("www." + ZONE1.dnsName(), DnsRecord.Type.AAAA)
+      .records(ImmutableList.of("ed:ed:12:aa:36:3:3:105"))
+      .ttl(25, TimeUnit.SECONDS)
+      .build();
+  public static final ChangeRequest CHANGE_ADD_ZONE1 = ChangeRequest.builder()
+      .add(A_RECORD_ZONE1)
+      .add(AAAA_RECORD_ZONE1)
+      .build();
+  public static final ChangeRequest CHANGE_DELETE_ZONE1 = ChangeRequest.builder()
+      .delete(A_RECORD_ZONE1)
+      .delete(AAAA_RECORD_ZONE1)
+      .build();
+
+  public static void purge() {
```

Comment:
Nit, I slightly prefer `clear` but feel free to leave as is if you think otherwise.